docs/modules/ROOT/pages/vshn-managed/postgresql/restore.adoc

```diff
@@ -0,0 +1,78 @@
+= Restore a PostgreSQL service
+
+IMPORTANT: The restores can only be triggered from an instance in the same namespace.
+If the old instance has already been deleted the backups are deleted as well.
+
+. Get the name of the backup you want to restore by listing all available backups.
++
+[source,bash]
+----
+kubectl -n postgres get vshnpostgresbackups
+BACKUP NAME                                 DATABASE INSTANCE     STATUS     AGE
+pgsql-app1-prod-pk8k4-2023-03-05-13-05-00   pgsql-app1-prod       Completed  38s
+pgsql-app1-dev-pk8k4-2023-03-01-16-52-11    pgsql-app1-dev        Failed     28s
+----
+NOTE: The list will show all backups from all instances in this namespace. In the example above there are two database instances with one backup each.
+
+. Choose a specific backup and get its details.
++
+[source,bash]
+----
+kubectl -n postgres get vshnpostgresbackups pgsql-app1-prod-pk8k4-2023-03-05-13-05-00 -oyaml
+----
++
+[source,yaml]
+----
+apiVersion: api.appcat.vshn.io/v1
+kind: VSHNPostgresBackup
+metadata:
+  name: pgsql-app1-prod-pk8k4-2023-03-05-13-05-00
+  namespace: postgres
+status:
+  databaseInstance: pgsql-app1-prod
+  process:
+    failure: "" <1>
+    jobPod: pgsql-app1-prod-pk8k4-backup-27967024-6xjpq
+    status: Completed
+    timing:
+      end: "2023-03-05T13:05:15.61228Z"
+      start: "2023-03-05T13:05:09.519369Z"
+      stored: "2023-03-05T13:05:15.755Z"
+----
+<1> The failure message in case the job fails.
+
+. Apply the following object on your namespace, as specified by its YAML description.
++
+.Example to restore a PostgreSQL instance
+[source,yaml]
+----
+apiVersion: vshn.appcat.vshn.io/v1
+kind: VSHNPostgreSQL
+metadata:
+  name: pgsql-app1-restore
+  namespace: postgres
+spec:
+  parameters:
+    restore:
+      claimName: pgsql-app1-prod <1>
+      backupName: pgsql-app1-prod-pk8k4-2023-03-01-16-52-02 <2>
+    backup:
+      schedule: '0 22 * * *'
+    service:
+      majorVersion: "15"
+      pgSettings:
+        timezone: Europe/Zurich
+    size:
+      cpu: "600m"
+      memory: "3500Mi"
+      disk: "80Gi"
+  writeConnectionSecretToRef:
+    name: postgres-creds-restored
+  compositionRef:
+    name: vshnpostgresrestore.vshn.appcat.vshn.io <3>
+----
+<1> The name of the instance you want to restore from
+<2> The backup name you want to restore
+<3> You need to specify that this is not a normal instance but a restored one
```

Comment:
Can I remove that again after the restore? Maybe the same questions as the next one for line 78.

Reply:
> Can I remove that again after the restore? Maybe the same questions as the next one for line 78.
By remove you mean edit the claim and erase the restore part?

Reply:
See above, it might be possible to switch the composition and thus then also be able to remove the restore part. But we probably need to investigate if that actually works or not...

Reply:
We somehow have to be able to restore the original object again...

Reply:
With the current restore strategy that we defined that's not possible. A restore is always into a new instance. https://products.docs.vshn.ch/products/appcat/data_protection.html#_restore_process

Reply:
We might want to move the restore object from `spec` to `status` after the restore process finished. I need to check  that to see if it's feasible. Will it work for you that way? Otherwise we will have to use composition functions.

Reply:
I don't think this is what I meant. I meant:

1. Specify the resource as described here, with all the needed parameters for the restore
2. Once the restore is finished: Remove all the restore related parameters again: `spec.parameters.restore` and `spec.compositionRef`.

Oh, and `spec.writeConnectionSecretToRef` is also a problem now, because it has a different name.

Put yourself into a users' perspective: You'd have to restore from a backup and after that, you have to configure your application for a different parameter and have a "tainted" resource in your deployment pipeline. Just doesn't feel good.

Reply:
An idea: We could change the restore process to:

1. Create a new restore instance, just like described here
2. Instruct the user to do some database dumping from the restore instance to the original instance
3. Remove the restore instance again

That could probably work as a suitable solution.

Reply:
This is very confusing. I would investigate a solution with Composition Functions and remove the restore composition altogether. We already tested it and it seems to work though it's in alpha for now.

Reply:
To summarize the issues here and that we're sure that we talk about the same thing.

There are two issues with this:

* The restored instance should permanently use a different `compositionRef`. We should figure out if we can change it to the normal `compositionRef` after the restore. This probably should just work.
* The documentation isn't clear enough that restoring will create a **new** instance and not revert the old one.

There are also related discussion points, but these aren't blocking but input we should look into

* Maybe we should provide a way to restore/revert the original instance
* Using different `compositionRef` is ugly, we should find a way around that. Maybe with composition functions

Did I understand both sides correctly?

Reply:
Yes, from my side. And: When the user restores into a new instance, the old instance would have to be deleted to save costs. But with the deletion, also older backups are gone, which is an issue. We should find a way to keep older backups, maybe transfer them to the new instance? Or, as written above, find a way to restore to the original instance.

Reply:
A follow-up work item APPCAT-382 as well as 2 new feature ideas in the product board are created. 
Shall we work on the documentation together in a separate session, @tobru ?